flink-connector-aws-e2e-tests/flink-connector-aws-kinesis-streams-e2e-tests/src/test/java/org/apache/flink/connector/kinesis/sink/KinesisStreamsSinkITCase.java

```diff
@@ -0,0 +1,357 @@
+import org.apache.flink.api.common.serialization.SimpleStringSchema;
+import org.apache.flink.connector.aws.testutils.AWSServicesTestUtils;
+import org.apache.flink.connector.aws.util.AWSGeneralUtil;
+import org.apache.flink.connector.kinesis.testutils.AWSEndToEndTestUtils;
+import org.apache.flink.connector.kinesis.testutils.AWSKinesisResourceManager;
+import org.apache.flink.runtime.client.JobExecutionException;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.source.datagen.DataGeneratorSource;
+import org.apache.flink.streaming.api.functions.source.datagen.RandomGenerator;
+import org.apache.flink.test.junit5.MiniClusterExtension;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.services.kinesis.KinesisClient;
+import software.amazon.awssdk.services.kinesis.model.GetRecordsRequest;
+import software.amazon.awssdk.services.kinesis.model.GetShardIteratorRequest;
+import software.amazon.awssdk.services.kinesis.model.ShardIteratorType;
+
+import java.util.Properties;
+
+import static org.apache.flink.connector.aws.config.AWSConfigConstants.AWS_ENDPOINT;
+
+/** End-to-end tests for Kinesis Data Streams Sink. */
+@ExtendWith(MiniClusterExtension.class)
+@Execution(ExecutionMode.CONCURRENT)
+@Tag("requires-aws-credentials")
+class KinesisStreamsSinkITCase {
+
+    private static final String DEFAULT_FIRST_SHARD_NAME = "shardId-000000000000";
+
+    private final SerializationSchema<String> serializationSchema = new SimpleStringSchema();
+    private final PartitionKeyGenerator<String> partitionKeyGenerator =
+            element -> String.valueOf(element.hashCode());
+    private final PartitionKeyGenerator<String> longPartitionKeyGenerator = element -> element;
+
+    private StreamExecutionEnvironment env;
+    private SdkHttpClient httpClient;
+    private KinesisClient kinesisClient;
+    private AWSKinesisResourceManager kinesisResourceManager;
+
+    @BeforeEach
+    void setUp() {
+        env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(1);
+
+        httpClient = AWSServicesTestUtils.createHttpClient();
+        kinesisClient =
+                AWSEndToEndTestUtils.createAwsSyncClient(httpClient, KinesisClient.builder());
+        kinesisResourceManager = new AWSKinesisResourceManager(kinesisClient);
+    }
+
+    @AfterEach
+    void teardown() {
+        kinesisResourceManager.close();
+        AWSGeneralUtil.closeResources(httpClient, kinesisClient);
```

Comment:
Should we run `cleanUpStaleKinesisDataStreams();` on `@AfterClass` as to speed up the tests? And possibly on `@BeforeClass` incase the tests starts failing...

Reply:
Thats a good idea. As discussed offline - moving this to a separate "dummy" test class